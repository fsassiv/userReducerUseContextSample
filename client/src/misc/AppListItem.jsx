import React from "react";
import "./AppListItem.scss";

const AppListItem = ({ item, itemTheme }) => {
  return (
    <a
      href={item.url}
      target="_blank"
      rel="noopener noreferrer"
      className={`list__item ${itemTheme}`}
    >
      <img src={item.image[1]["#text"]} alt="" className="list__item-cover" />
      <div className="list__item-info">
        <p className="list__item-label">{item.name}</p>
        {item.listeners ? (
          <span className="list__item-stats">{item.listeners} ouvintes</span>
        ) : (
          <span className="list__item-stats">{item.artist}</span>
        )}
      </div>
    </a>
  );
};

export default AppListItem;
