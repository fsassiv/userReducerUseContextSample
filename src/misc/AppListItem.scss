.list__item {
  height: 60px;
  width: 100%;
  display: -webkit-box;
  display: flex;
  -webkit-box-pack: justify;
  justify-content: space-between;
  -webkit-box-align: center;
  align-items: center;
  border-width: 1px;
  border-style: solid;
  margin-bottom: 5px;
  padding: 0 10px;
  &-cover {
    width: 40px;
    height: 40px;
    border-radius: 50%;
    -o-object-fit: cover;
    object-fit: cover;
  }
  &-info {
    width: 80%;
  }
  &-label {
    font-weight: bold;
    font-size: 14px;
  }
  &-stats {
    font-size: 12px;
    font-weight: 300;
  }
  &.light {
    border-color: black;
    background-color: white;
    color: black;
  }
  &.dark {
    border-color: white;
    color: white;
    background-color: black;
  }
}
