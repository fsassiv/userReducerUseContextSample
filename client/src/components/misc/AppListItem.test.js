import AppListItem from "./AppListItem.jsx";

describe("AppListItem", () => {
  const props = {
    item: { image: ["fist", "second"], url: "some http", name: "item name" },
    itemTheme: "dark"
  };

  const appListItem = shallow(<AppListItem {...props} />);

  it("rendered correctly", () => {
    expect(appListItem).toMatchSnapshot();
  });

  it("check if it has the right class", () => {
    expect(appListItem.find(".list__item").hasClass("dark")).toBeTruthy();
  });
});
