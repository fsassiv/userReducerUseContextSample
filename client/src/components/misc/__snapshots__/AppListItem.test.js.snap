// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppListItem rendered correctly 1`] = `ShallowWrapper {}`;
